### DISCUSSION TITLE
Removal of the node's edge removes the placeholder for the operator

### DISCUSSION THREAD

**Discussion by hubertp** [enso-org/enso#6907] [Work In Progress - GUI]
I'm not sure if there is any easy to way to remedy this but just wanted to bring this to the attention of UX.
Removing the edge removes the `_` placeholder and the only way to bring it back is to add it manually.
Maybe there are better ways to deal with it?

[Kazam_screencast_00011.webm](https://github.com/enso-org/enso/assets/292128/4a6374e0-c574-419b-b358-ef819afb9fa6)

**Comment by farmaazon** [comment]
Technically, we could add a similar mechanism to adding elements to a list. But so far there was no much demand for that.

**Reply by wdanilo** [reply]
Hmm, but when removing connection, we are normally leaving `_` there, why here we are changing the expression and removing the `+` as well? I'm not sure what is the correct/good way of doing it yet, I'm just trying to understand what is the current logic there.

**Reply by hubertp** [reply]
> Hmm, but when removing connection, we are normally leaving `_` there, why here we are changing the expression and removing the `+` as well? 

Yes, that was my expectation as well.

**Reply by farmaazon** [reply]
> Hmm, but when removing connection, we are normally leaving _ there

Not quite. Reconnecting has currently a logic of _removing_ target port. We leave `_` only when it's impossible without losing information needed to connect to the port again: for example, when you remove the second-last operand in some `+` (or whatever operator) chain (`.` also counts as operator - old UFCS remains. That's why disconnecting self port leaves `_`). And actually, this is the _only_ case I know.

So, as you see, this logic has an assumption that there's no problem in restoring a removed element, and SpanTree has logic needed to do that: just the views never show the proper "insertion" ports.

**Reply by wdanilo** [reply]
Ok, @hubertp @farmaazon @sylwiabr – according to your feelings from user-perspective, purely UX perspective, how should it behave in your opinions?

**Reply by kazcw** [reply]
Right now, when the edge is clicked on the target side (to disconnect the source end), the target port is removed from the expression and we're left holding an edge to nowhere. In that case at least, I think we should keep a placeholder so that the edge is still valid and can be attached to a new source.

**Reply by wdanilo** [reply]
@kazcw I believe that @Frizi is right now chaining this behavior in such a way, that if you disconnect the end of the edge, the code is not changed until you drop the edge / connect it somewhere else. @Frizi can you confirm it, please?

**Reply by Frizi** [reply]
Yes, I'm currently working on that. The goal is such that when the edge is clicked on the target side, the code will remain unchanged until you do something with that edge (either connect or drop). The behavior will be unchanged when you are holding an edge connected to a source port.

I believe that the original issue could be adressed by having an insertion point inbetween chain elements, similar to how we have it right now in lists. Though making it look good and intuitive is probably another whole thing.

**Comment by farmaazon** [comment]
Closing stale discussion. I propose to _not_ implement removing operands from operator chains, and see if we actually need it.
